File: backend/qms-system/src/main/java/nz/ac/autuni/qmssystem/model/FMEATableGroup.java
package nz.ac.autuni.qmssystem.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.List;

/**
 * @author Simon-the-coder
 * @date 21/11/20 11:16 am
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Document(collection = "FMEATableGroup")
public class FMEATableGroup implements Serializable {
    @Id
    String fmeaTableGroupId;
    String riskProcedureId;
    List<FMEATable> fmeaTableList;
}

File: frontend/QMS_system/lib/model/fmea_table_group.dart
import 'package:QMS_system/model/fmea_table.dart';

class FMEATableGroup {
  String fmeaTableGroupId;
  String riskProcedureId;
  List<FMEATable> fmeaTableList;

  FMEATableGroup({
    this.fmeaTableGroupId,
    this.riskProcedureId,
    this.fmeaTableList,
  });
  FMEATableGroup.fromJson(Map<String, dynamic> json)
      : fmeaTableGroupId = json['fmeaTableGroupId'],
        riskProcedureId = json['riskProcedureId'],
        fmeaTableList = json['fmeaTableList'].map<FMEATable>((json) => FMEATable.fromJson(json)).toList(growable: false);

  Map<String, dynamic> toJson() => {
    'fmeaTableGroupId' : fmeaTableGroupId,
    'riskProcedureId': riskProcedureId,
    'fmeaTableList': fmeaTableList.map<Map<String,dynamic>>((item) => item.toJson()).toList(growable: false),
  };
}